src/App.jsx

```diff
@@ -14,33 +15,116 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+export const App = () => {
+  const [goods, setGoods] = useState(goodsFromServer);
+  const [sortedAlph, setSortedAlph] = useState(false);
+  const [sortedByLength, setSortedByLength] = useState(false);
+  const [reverse, setReverse] = useState(false);
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+  const applySortingAndReverse = (array, sortType) => {
+    const updatedGoods = [...array];
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+    if (sortType === 'alphabetical') {
+      updatedGoods.sort((good1, good2) => good1.localeCompare(good2));
+    } else if (sortType === 'length') {
+      updatedGoods.sort((good1, good2) => good1.length - good2.length);
+    }
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+    if (reverse) {
+      updatedGoods.reverse();
+    }
+
+    return updatedGoods;
+  };
+
+  const changeGoods = condition => {
+    switch (condition) {
+      case 'sortAlph':
+        if (sortedAlph) {
+          setSortedAlph(false);
+          setGoods(applySortingAndReverse(goodsFromServer, null));
```

Comment:
Passing `null` as the `sortType` to `applySortingAndReverse` may lead to unexpected behavior since the function does not handle a `null` sort type. Consider adding a condition to handle this case within the function.